### PR TITLE
Add FXIOS-5692 [v112] Credit card settings UX - Add accessibility for credit card list

### PR DESCRIPTION
**Pull request by thatswinnie**
[FXIOS-5692](https://mozilla-hub.atlassian.net/browse/FXIOS-5692)

This adds the autofill toggle to the empty list view, updates fonts to use dynamic fonts and add labels for voice over.

### CODE REVIEW

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 29.14%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 15 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 4 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 25.73
| File | Coverage ||
| --- | --- | --- |
Style.swift | 0.0% | ⚠️
CreditCardSettingsViewController.swift | 0.0% | ⚠️
CreditCardItemRow.swift | 0.0% | ⚠️
CreditCardSettingsViewModel.swift | 0.0% | ⚠️
CreditCardSettingsEmptyView.swift | 0.0% | ⚠️
AdaptiveStack.swift | 0.0% | ⚠️
RemoveCardButton.swift | 0.0% | ⚠️
CreditCardSectionHeader.swift | 0.0% | ⚠️
FloatingTextField.swift | 0.0% | ⚠️
PreferredFont.swift | 0.0% | ⚠️
CreditCardTableViewController.swift | 0.0% | ⚠️
CreditCardAutofillToggle.swift | 0.0% | ⚠️
InstructionsView.swift | 0.0% | ⚠️
CreditCardTableViewModel.swift | 0.0% | ⚠️
HostingTableViewSectionHeader.swift | 0.0% | ⚠️


## ShareTo.appex: Coverage: 30.83
| File | Coverage ||
| --- | --- | --- |
InstructionsView.swift | 0.0% | ⚠️


## WidgetKitExtension.appex: Coverage: 22.62
| File | Coverage ||
| --- | --- | --- |
TopSitesWidget.swift | 0.0% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 3c927cfce3cc9f6ba7ed452b28664971fab10ede
</p>

File: Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsEmptyView.swift [hunk 1]
```diff
@@ -6,39 +6,62 @@ import Foundation
...
-                    .padding([.top], -5)
-                Spacer()
+            GeometryReader { proxy in
+                ScrollView {
+                    VStack {
+                        CreditCardAutofillToggle(
+                            textColor: colors.toggleTextColor,
+                            model: toggleModel)
+                        Spacer()
+                        Image("credit_card_placeholder")
```
Comment (nishant2718) [inline Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsEmptyView.swift:29]: nit: Maybe we can switch it back to `ImageIdentifiers.creditCardPlaceholder` if it's the same

File: Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardTableViewController.swift [hunk 2]
```diff
@@ -145,18 +91,13 @@ class CreditCardTableViewController: UIViewController, Themeable {
...
     func applyTheme() {
         let theme = themeManager.currentTheme
-        toggleSwitchContainerLine.backgroundColor = theme.colors.borderPrimary
-        toggleSwitchContainer.backgroundColor = theme.colors.layer2
-        tableView.backgroundColor = .clear
-        toggleSwitch.onTintColor = theme.colors.actionPrimary
+//        toggleSwitchContainerLine.backgroundColor = theme.colors.borderPrimary
+//        toggleSwitchContainer.backgroundColor = theme.colors.layer2
+//        tableView.backgroundColor = .clear
+//        toggleSwitch.onTintColor = theme.colors.actionPrimary
```
Comment (nishant2718) [inline Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardTableViewController.swift:97]: nit: these can be removed if we don't need them

File: Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardAutofillToggle.swift [hunk 3]
```diff
@@ -15,16 +34,34 @@ struct CreditCardAutofillToggle: View {
                 .padding(.leading, 16)
                 .hidden()
             HStack {
-                Toggle(isOn: $toggleState) {
-                    Text("Save and autofill cards")
-                }.font(.system(size: 17))
-                 .padding(.leading, 16)
-                 .padding(.trailing, 16)
+                Toggle("Save and autofill cards", isOn: $model.isEnabled)
```
Comment (nishant2718) [inline Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardAutofillToggle.swift:37]: nit: If it's not too much trouble, maybe we can change this string to `ToggleToAllowAutofillTitle`

**Review by nishant2718 — APPROVED** [review]
Some minor nits; looks good!

File: Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsEmptyView.swift (hunk 1 above)
Comment (nbhasin2) [inline Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsEmptyView.swift:29]:
```suggestion
                        Image(ImageIdentifiers.creditCardPlaceholder)
```

File: Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardTableViewController.swift (hunk 2 above)
Comment (nbhasin2) [inline Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardTableViewController.swift:97]:
```suggestion
```

File: Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardAutofillToggle.swift (hunk 3 above)
Comment (nbhasin2) [inline Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardAutofillToggle.swift:37]:
```suggestion
                Toggle(String.CreditCard.EditCard.ToggleToAllowAutofillTitle, isOn: $model.isEnabled)
```

**Comment by nbhasin2** [comment]
Made the changes while @thatswinnie is away :) 
Going to run BR once more before merging @nishant2718

**Pull request merged**